File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/GetAllWordsUseCase.java
package com.przemyslawlusnia.vocabularycreator.wordlist.domain;

import com.przemyslawlusnia.vocabularycreator.core.RxUseCase;
import com.przemyslawlusnia.vocabularycreator.wordlist.repository.WordsRepository;
import java.util.List;
import rx.Observable;
import rx.Scheduler;

public class GetAllWordsUseCase extends RxUseCase<List<WordDomainModel>> {

  private WordsRepository wordsRepository;

  public GetAllWordsUseCase(Scheduler executionScheduler, WordsRepository wordsRepository) {
    super(executionScheduler);
    this.wordsRepository = wordsRepository;
  }

  @Override
  protected Observable<List<WordDomainModel>> buildUseCaseObservable() {
    return wordsRepository.getAllWords();
  }
}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/GetAllWordsUseCase.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.domain

import com.przemyslawlusnia.vocabularycreator.core.RxUseCase
import com.przemyslawlusnia.vocabularycreator.wordlist.repository.WordsRepository
import rx.Observable
import rx.Scheduler

class GetAllWordsUseCase(executionScheduler: Scheduler, private val wordsRepository: WordsRepository)
    : RxUseCase<List<WordDomainModel>>(executionScheduler) {

    public override fun buildUseCaseObservable(): Observable<List<WordDomainModel>> {
        return wordsRepository.allWords
    }
}
